types/artifacts.d.ts

```diff
@@ -189,6 +190,10 @@ declare global {
     }
 
     module Artifacts {
+      export type ComputedContext = Immutable<{
```

Comment:
Kind of a tangent for where this should live now, this is the kind of thing I think is a good argument for us moving away from global types (no one needs to reference this type except computed artifacts, which already need to import `ComputedArtifact`, so that's a good namespace for the type to live in), but typescript really needs to fix https://github.com/microsoft/TypeScript/issues/41825 before that's a very ergonomic approach for us in general.